sympy/tensor/array/ndim_array.py

```diff
@@ -140,10 +140,7 @@ def _handle_ndarray_creation_inputs(cls, iterable=None, shape=None, **kwargs):
             # Construction of a sparse array from a sparse array
             elif isinstance(iterable, SparseNDimArray):
                 return iterable._shape, iterable._sparse_array
-            # Construction from another `NDimArray`:
-            elif isinstance(iterable, NDimArray):
-                shape = iterable.shape
-                iterable = list(iterable)
```

Comment:
why was this removed???

Reply:
Because we are trying to cast an `Array` to `list` and then flatten it in the constructor with `flatten()`. Since `flatten()` can now flatten an `Array` properly, I was thinking that we don't need this case any more.

Reply:
OK